### ISSUE TITLE
Constructor's Parameter hints are still not working. (MSVC)

### ISSUE THREAD

**Issue by JoeyRxy** [microsoft/vscode-cpptools#7258]
**Type: LanguageService**

I found an old issue #819 which claims the bug is solved, but I still encounter this problem. I found the bug is fixed for most functions, but not the Constructors. We know that constructors in c++ is called when we type the parentheses after the object name, like `vector<int> v(10, -1);`. When I typed the left parentheses, the parameter hints of the constructors didn't pop up. But after I type the first parameter and a comma, the hints pop up. like the gif below:
![vscode-cpp-bug-gif](https://raw.githubusercontent.com/JoeyRxy/tmp/master/vscode-cpp-problem.gif)

**Describe the bug**
- OS and Version: win 10 (2004)
- VS Code Version: 1.54.3
- C/C++ Extension Version: v1.3.0-insiders2
- Other extensions you installed (and if the issue persists after disabling them): Material Icon theme, settings sync. And yes, issue persists.
- Does this issue involve using SSH remote to run the extension on a remote machine?:No
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).
My workstation just contains one c++ file

**Steps to reproduce**

1. type `vector<int> a(` in a cpp file.
2. after typing the left parentheses, the hints of constructor will not pop up.
3. The hints will not pop up until you type the first parameter and a comma, like `10` and `,`.

**Expected behavior**

the parameter hints should pop up after typing the parentheses, because at this time, the language server should recognize this is a constructor.

1. Logs from the command `C/C++: Log Diagnostics`
None
2. Logs from [the language server](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)
After typing the left parentheses:
```log
sending 1 changes to server
Checking for syntax errors: file:///c%3A/Users/Rxy/Desktop/project/cpp/cpp_primer/test3.cpp
Queueing IntelliSense update for files in translation unit of: C:\USERS\RXY\DESKTOP\PROJECT\CPP\CPP_PRIMER\TEST3.CPP
  tag parsing file: C:\USERS\RXY\DESKTOP\PROJECT\CPP\CPP_PRIMER\TEST3.CPP
Error squiggle count: 0
Update IntelliSense time (sec): 2.267
```
Then type first parameter and a comma, the log appends:
```log
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
Checking for syntax errors: file:///c%3A/Users/Rxy/Desktop/project/cpp/cpp_primer/test3.cpp
Queueing IntelliSense update for files in translation unit of: C:\USERS\RXY\DESKTOP\PROJECT\CPP\CPP_PRIMER\TEST3.CPP
  tag parsing file: C:\USERS\RXY\DESKTOP\PROJECT\CPP\CPP_PRIMER\TEST3.CPP
Error squiggle count: 1
Update IntelliSense time (sec): 2.705
```

**Screenshots**

![vscode-cpp-bug-gif](https://raw.githubusercontent.com/JoeyRxy/tmp/master/vscode-cpp-problem.gif)
**Additional context**

**Comment by JoeyRxy** [comment]
@bobbrow hope you can spare time viewing this issue. Thanks for your valuable time.

**Comment by sean-mcmanus** [comment]
There is a known ambiguity with C++'s parsing of declarations like "type name();" because it could be a foward declaration of a function "name" that takes no arguments and returns a "type" (e.g. "int mySum();"). It is recommended that you use "type name{}" syntax to resolve this ambiguity, in which case signature help works correctly.

However, this still might be a bug (I get different results with VS, although VS could have a bug too).

**Comment by sean-mcmanus** [comment]
I got a repro using Visual Studio using
```
struct foo
{
    foo(){}
    foo(int i){}
};


int main()
{
    foo f1();
    foo f2{};
    return 0;
}
```
so I filed a bug report on them (i.e. shared code) at https://developercommunity.visualstudio.com/t/C-signature-help-doesnt-work-for-cons/1386843.

**Comment by sean-mcmanus** [comment]
Another workaround is to set editor.autoClosingBrackets to "never" or delete the closing ) and type Ctrl+Shift+Space (invoke signature help).

The issue is also being tracked at https://developercommunity2.visualstudio.com/t/C-function-parameter-hints-for-constru/676371

**Comment by JoeyRxy** [comment]
> There is a known ambiguity with C++'s parsing of declarations like "type name();" because it could be a foward declaration of a function "name" that takes no arguments and returns a "type" (e.g. "int mySum();"). It is recommended that you use "type name{}" syntax to resolve this ambiguity, in which case signature help works correctly.
> 
> However, this still might be a bug (I get different results with VS, although VS could have a bug too).

Thanks, use `{}` works fine. I'll get used to this way to init.

**Comment by sean-mcmanus** [comment]
This is fixed in our latest 1.11.5 release (maybe an earlier one too).
